### ISSUE TITLE
Constants?

### ISSUE THREAD

**Issue by piegamesde** [SergiusIW/noisy_float-rs#32]
As far as I can tell, none of the functions are `const`, but this is required in order to define some constants. I could use `lazy_static`, but this wouldn't be a zero-cost abstraction.

**Comment by SergiusIW** [comment]
The main function that would benefit from const is NoisyFloat::new(...), or the similar helper functions r64, n64, etc. I need const panics for that to be able to work properly, which are not yet implemented (see https://github.com/rust-lang/rust/issues/51999). With that in place, I could maybe get the r64/n64/r32/n32 working as const functions. I would also need const functions on traits to get the whole thing working, not sure what the current state of that is in Rust.

**Comment by piegamesde** [comment]
If it's only the panicking, maybe make some unsafe constructor const?

**Comment by SergiusIW** [comment]
Tried to make NosiyFloat::unchecked_new(...) const and public, but ran into the following error:
```
error[E0723]: trait bounds other than `Sized` on const fn parameters are unstable
   --> src/lib.rs:156:6
    |
156 | impl<F: Float, C: FloatChecker<F>> NoisyFloat<F, C> {
    |      ^
    |
    = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
    = help: add `#![feature(const_fn)]` to the crate attributes to enable
```

If you would like, I could still implement an unchecked const constructor specifically for the R32, R64, N32, and N64 types, but the general const constructor will not work yet, not until this feature hits stable Rust.

**Comment by piegamesde** [comment]
Yes, I'd really appreciate that. It doesn't look like the required features will be stabilized anytime soon

**Comment by SergiusIW** [comment]
Ok, it's done. Just released version 0.1.14. The following now works:

```Rust
const A: N32 = N32::unchecked_new(1.0);
const B: N64 = N64::unchecked_new(2.0);
const C: R32 = R32::unchecked_new(3.0);
const D: R64 = R64::unchecked_new(4.0);

const A_RAW: f32 = A.const_raw();
const B_RAW: f64 = B.const_raw();
const C_RAW: f32 = C.const_raw();
const D_RAW: f64 = D.const_raw();

assert_eq!(A_RAW, 1.0);
assert_eq!(B_RAW, 2.0);
assert_eq!(C_RAW, 3.0);
assert_eq!(D_RAW, 4.0);
```
